libass/ass.c

```diff
@@ -1235,6 +1235,31 @@ static char *sub_recode(ASS_Library *library, char *data, size_t size,
 }
 #endif                          // ICONV
 
+FILE* ass_fopen(const char* _FileName, const char* _Mode) {
+#if defined(_WIN32) && !defined(__CYGWIN__)
+    wchar_t* wpath, * wmode;
+    UINT cur_cp = CP_UTF8;
+    FILE* f = NULL;
+    wpath = ass_convert_to_utf16le(_FileName, cur_cp), wmode = ass_convert_to_utf16le(_Mode, cur_cp);
```

Comment:
Use of an universal conversion function for the file mode looks like a great overkill for me. I've used this code in a similar context:
```c
wchar_t wmode[4] = L"\0b\0";
while (*mode) switch (*mode++) {
    case 'r':  wmode[0] = L'r';  break;
    case 'w':  wmode[0] = L'w';  break;
    case 'a':  wmode[0] = L'a';  break;
    case '+':  wmode[2] = L'+';  break;
}
```
But even that is unneeded for libass, we only use "rb". So it's better to drop the second argument of `ass_fopen()` completely.

Reply:
I admit this and will change it in a later commit.